### PR TITLE
Fix pull request triggers from forks

### PR DESCRIPTION
**Pull request by ch3lo**
This PR fix the problem when the PR hook is triggered from forks. https://github.com/Nerdwin15/stash-jenkins-postreceive-webhook/issues/96

Today when the PR Trigger Button is clicked or a hook is dispatched in background (new PR, Reopen, change....), the plugin notifies Jenkins with the from sha1 and the from branch, but the last is useless if you are trying to reintegrate a PR or test your integration, because set a `*`, `**`, wherever.. in **Branch Specifier** field could be ambiguous and not clarify if it is a PR or not.

Additionally I added a new parameter in notification called `TARGET_BRANCH`, it is useful in merge when destination branch is not master or a default branch. It needs git plugin 2.4+.

This PR notifies always the branch parameter in this format: `pr/<pull request id>/from`. I used `from` because, `merge` ref is **async** https://github.com/Nerdwin15/stash-jenkins-postreceive-webhook/issues/83#issuecomment-60468105 and if you need to merge your code before test, it could be made easy in Jenkins with:

Additional Behaviours -> Merge Before Build with:
Name of repository: origin
Branch to merge to: $TARGET_BRANCH

With this PR, your Jenkins Git config should look like:
Name: origin
Refspec: +refs/pull-requests/_/from:refs/remotes/origin/pr/_
Branch Specifier: origin/pr/*/from

sorry about my english

Signed-off-by: Marcelo Salazar chelosalazar@gmail.com

### CODE REVIEW

**Comment by go2sh** [comment]
I like your pull requests. Can't you add parameters to that url as it is not supported by the git plugin?

**Comment by ch3lo** [comment]
I don't know if you can add additional parameters previous git 2.4 version, this is the reason why I added a omit parameters.

**Comment by mattadamson** [comment]
Hi Marcelo, I'm very excited to see this PR. Does this mean you have this working in stash with a single Jenkins job for forked repositories? We typically merge from a fork for each developer and also in a separate feature / bug fix branch back to master or the central release branch. 

I commented here

https://github.com/Nerdwin15/stash-jenkins-postreceive-webhook/issues/96

**Comment by elee** [comment]
This PR is relevant to my interests. Is there any barrier to merging this?

File: src/main/java/com/nerdwin15/stash/webhook/Notifier.java [hunk 1]
```diff
@@ -177,21 +186,25 @@ public NotificationResult call() throws Exception {
...
    *        in notification to Jenkins.
    * @param omitBranchName Defines whether the commit's branch name is omitted
+   * @param omitTargetBranch Omit the target branch
    * @return The notification result.
    */
   public @Nullable NotificationResult notify(@Nonnull Repository repo, //CHECKSTYLE:annot
       String jenkinsBase, boolean ignoreCerts, String cloneType, String cloneUrl,
-      String strRef, String strSha1, boolean omitHashCode, boolean omitBranchName) {
+      String strRef, String strSha1, String targetBranch, boolean omitHashCode, 
+      boolean omitBranchName, boolean omitTargetBranch) {
```
Comment (alexBraidwood) [inline src/main/java/com/nerdwin15/stash/webhook/Notifier.java]: I feel the omit bools aren't clear enough. I think we should prefer names like shouldOmitTargetBranch. To me, it makes statements like `if (!shouldOmitTargetBranch)` more clear.

File: src/main/java/com/nerdwin15/stash/webhook/PullRequestEventListener.java [hunk 2]
```diff
@@ -62,16 +62,16 @@ protected void handleEvent(PullRequestEvent event) {
       return;
     }
 
-    String strRef = event.getPullRequest().getFromRef().toString()
-        .replaceFirst(".*refs/heads/", "");
+    String strRef  = "pr/" + Long.toString(event.getPullRequest().getId()) + "/from";
```
Comment (alexBraidwood) [inline src/main/java/com/nerdwin15/stash/webhook/PullRequestEventListener.java]: What are the implications of this change? It looks we're changing the whole PR scheme here.

File: src/main/java/com/nerdwin15/stash/webhook/Notifier.java (hunk 1 above)
Comment (ch3lo) [inline src/main/java/com/nerdwin15/stash/webhook/Notifier.java]: yes, I wrote in that manner to maintain consistency with the others vars.

File: src/main/java/com/nerdwin15/stash/webhook/PullRequestEventListener.java (hunk 2 above)
Comment (ch3lo) [inline src/main/java/com/nerdwin15/stash/webhook/PullRequestEventListener.java]: If I remember well, i'ts necessary, because a PR can come from a fork.

Comment (alexBraidwood) [inline src/main/java/com/nerdwin15/stash/webhook/PullRequestEventListener.java]: Alright, I'm going to test this out and look into getting it in.

**Comment by alexBraidwood** [comment]
Pulled this in, I didn't run into any problems while testing on 4.0.1 branch.

**Pull request merged**